src/Generator.php

```diff
@@ -170,10 +170,14 @@ private function addOperation(string $targetPath, string $templateName, array $v
 
         $templatePath = $templateName;
         if (!file_exists($templatePath)) {
-            $templatePath = __DIR__.'/Resources/skeleton/'.$templateName;
+            $templatePath = dirname(__DIR__, 4)."/src/Maker/Resources/skeleton/".$templateName;
```

Comment:
The value of the parameter `%kernel.project_dir%` should  be used  instead of `dirname(__DIR__, 4)`.

Also, it will be great if we find (if possible) a way to not hardcode this path `/src/Maker/Resources/skeleton/`.

Reply:
When I tried with `%kernel.project_dir%` I got syntax errors, I tried with double quoted `PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: syntax error, unexpected '%' in /media/sf_sskeleton/vendor/symfony/maker-bundle/src/Generator.php:176`

Why not hardcode `/src/Maker/Resources/skeleton/`? if path doesn't exists then will be vendor directory searched.

Reply:
I changed `dirname(__DIR__, 4)`. there is `getRootDirectoty()` function few lines above :)

Reply:
Yes, `getRootDirectory()` is what I meant with `The value of the parameter %kernel.project_dir%`.

> Why not hardcode /src/Maker/Resources/skeleton/? if path doesn't exists then will be vendor directory searched. This is normal behavior in other bundles, for example in Twig. You can overwrite each template in vendor directory creating the same folder structure in src directory

Right, but it is hardcoding a convention, what if I want to store my skeleton templates in `src/Resources/MakerBundle/skeleton/` instead? or in `templates/maker/skeleton/`?

Configuration over convention, so I'm asking for a general solution like "a paths config/option" where to looking for skeleton templates.

Reply:
> Right, but it is hardcoding a convention, what if I want to store my skeleton templates in src/Resources/MakerBundle/skeleton/ instead? or in templates/maker/skeleton/?

Because of convetnion of symfony? even now this path `$templatePath = __DIR__.'/Resources/skeleton/'.$templateName;` is fully hardcoded and users creating own makers can't use own path in symfony way.

> Configuration over convention, so I'm asking for a general solution like "a paths config/option" where to looking for skeleton templates.

I also think that It would be better to store this path in some variable, but Maker-Bundle doesn't provide any configuration file or doesn't add any lines into .env file. So the simplest way to achevie oportunity to store maker templates in own project, not vendor directory.

Reply:
I tried to add variable in config but i get error ` Unrecognized option "template_path" under "maker". Available option is "root_namespace". `

Reply:
> For that reason, in general, the generated code cannot be modified. In many cases, adding your own maker command is so easy, that we recommend that.

I admit, adding own maker command is so easy, but not adding own templates without coding in vendor dir  :/

Reply:
> I also think that It would be better to store this path in some variable, but Maker-Bundle doesn't provide any configuration file ...

Look at this file https://github.com/symfony/maker-bundle/blob/master/src/DependencyInjection/Configuration.php

The `root_namespace` is already a config option for `Generator` service, what about add a new option `skeleton_path` (or better) to configure the default skeleton path at project level?